src/rgw/driver/sfs/sqlite/bindings/blob.h

```diff
@@ -13,28 +13,44 @@
  */
 #pragma once
 
+#include <tuple>
 #include <type_traits>
 
 #include "rgw/driver/sfs/sqlite/conversion_utils.h"
+// we need to include dbapi_type_wrapper.h only because including dbapi.h
+// creates circular dependencies
+#include "rgw/driver/sfs/sqlite/dbapi_type_wrapper.h"
 #include "rgw/driver/sfs/sqlite/sqlite_orm.h"
 #include "rgw_common.h"
 
-namespace sqlite_orm {
+namespace blob_utils {
 
-template <typename T>
-struct __is_sqlite_blob : std::false_type {};
+template <typename T, typename Tuple>
+struct has_type;
 
 template <typename T>
-inline constexpr bool is_sqlite_blob = __is_sqlite_blob<T>::value;
+struct has_type<T, std::tuple<>> : std::false_type {};
 
-template <>
-struct __is_sqlite_blob<rgw::sal::Attrs> : std::true_type {};
+template <typename T, typename U, typename... Ts>
+struct has_type<T, std::tuple<U, Ts...>> : has_type<T, std::tuple<Ts...>> {};
 
-template <>
-struct __is_sqlite_blob<ACLOwner> : std::true_type {};
+template <typename T, typename... Ts>
+struct has_type<T, std::tuple<T, Ts...>> : std::true_type {};
 
-template <>
-struct __is_sqlite_blob<rgw_placement_rule> : std::true_type {};
+// list of types that are stored as blobs and have the encode/decode functions
+using BlobTypes = std::tuple<
+    rgw::sal::Attrs, ACLOwner, rgw_placement_rule,
+    std::map<std::string, RGWAccessKey>, std::map<std::string, RGWSubUser>,
+    RGWUserCaps, std::list<std::string>, std::map<int, std::string>,
+    RGWQuotaInfo, std::set<std::string>, RGWBucketWebsiteConf,
+    std::map<std::string, uint32_t>, RGWObjectLock, rgw_sync_policy_info>;
```

Comment:
Should we add also: RGWBucketInfo? 
Saying this because I'm using encode/decode functions over this type but maybe not needed.

Reply:
If we store the whole bucket info as a blob we can't then query over specific bucket columns. That's why we didn't store buckets that way.

Reply:
right!